src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -799,6 +803,22 @@ impl Edges {
         detached.extend(detached_source_iter);
         detached.into_iter()
     }
+
+    pub fn node_attached_edges(&self, node_id:NodeId) -> Vec<EdgeId> {
+        self.all.raw.borrow().iter().filter_map(|(&id,edge)|{
```

Comment:
Do I understand correctly that you are iterating over ALL edges in the graph here and checking whether they are attahced to a particular node? If so, this is super, super slow. I believe that nodes remember their incoming / outcoming edges, so you should rather use that information instead of iterating over everytihng,

Reply:
Good point!